sklearn/compose/tests/test_target.py

```diff
@@ -163,12 +163,13 @@ def test_transform_target_regressor_2d_transformer(X, y):
     lr = LinearRegression()
     transformer2 = clone(transformer)
     if y.ndim == 1:  # create a 2D array and squeeze results
-        lr.fit(X, transformer2.fit_transform(y.reshape(-1, 1)).squeeze())
+        lr.fit(X, transformer2.fit_transform(y.reshape(-1, 1)))
     else:
         lr.fit(X, transformer2.fit_transform(y))
     y_lr_pred = lr.predict(X)
-    assert_allclose(y_pred, transformer2.inverse_transform(y_lr_pred))
-    assert_allclose(regr.regressor_.coef_, lr.coef_)
+    assert_allclose(y_pred,
+                    transformer2.inverse_transform(y_lr_pred).squeeze())
+    assert_allclose(regr.regressor_.coef_, lr.coef_.squeeze())
```

Comment:
I think this part can be clearer:

```python
    if y.ndim == 1:  # create a 2D array and squeeze results
        lr.fit(X, transformer2.fit_transform(y.reshape(-1, 1)).squeeze())
        y_lr_pred = lr.predict(X).reshape(-1, 1)
        y_pred2 = transformer2.inverse_transform(y_lr_pred).squeeze()
    else:
        lr.fit(X, transformer2.fit_transform(y))
        y_lr_pred = lr.predict(X)
        y_pred2 = transformer2.inverse_transform(y_lr_pred)

    assert_allclose(y_pred, y_pred2)
    assert_allclose(regr.regressor_.coef_, lr.coef_)
```

This way the reshaping we need for `y.ndim == 1` is done in the if statement, while the check for `regr` and `y_pred` can stay the same.

In the future PR, I think it would be good to break this test into two: one to handle 1d y and another to handle 2d y. (The `1d` case is where we have to do all the shaping.)

Reply:
Thanks a lot for the advice Thomas!